lib/rubyipmi/commands/basecommand.rb

```diff
@@ -37,9 +37,8 @@ def initialize(commandname, opts = ObservableHash.new)
     end
 
     def locate_command(commandname)
-      location = `which #{commandname}`.strip
-      unless $?.success?
-        logger.error("#{commandname} command not found, is #{commandname} installed?") if logger
+      unless Rubyipmi.locate_command(commandname)
+        logger&.error("#{commandname} command not found, is #{commandname} installed?")
         raise "#{commandname} command not found, is #{commandname} installed?"
       end
```

Comment:
While I was digging, I spotted this method looked the same as the class method so I felt like DRY.

Reply:
Definitely, probably many opportunities to refactor :)